### PR TITLE
IGVF-1170-disallow-empty-string-desc

### PR DESCRIPTION
**Pull request by meenakshikagda**

### CODE REVIEW

File: src/igvfd/schemas/changelogs/mixins.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Changelog for *`mixins.json`*
 
+* Disallow empty strings in `descriptions`. (12/02/2023)
```
Comment (ian-whaling) [inline src/igvfd/schemas/changelogs/mixins.md:3]: description not descriptions

Comment (gabdank) [reply src/igvfd/schemas/changelogs/mixins.md:3]: good catch

**Review by gabdank — CHANGES_REQUESTED** [review]
typo Ian found in mixins

File: src/igvfd/tests/fixtures/schemas/analysis_set.py [hunk 2]
```diff
@@ -71,3 +71,14 @@ def analysis_set_v3(award, lab):
...
     }
     return item
+
+
+@pytest.fixture
+def analysis_set_v4(analysis_set_v3):
+    item = analysis_set_v3.copy()
+    item.update({
+        'schema_version': '4',
+        'file_set_type': 'itermediate analysis',
```
Comment (ian-whaling) [inline src/igvfd/tests/fixtures/schemas/analysis_set.py:81]: don't think this line is necessary

File: src/igvfd/tests/test_upgrade_award.py [hunk 3]
```diff
@@ -13,3 +13,9 @@ def test_award_upgrade_2_3(upgrader, award_v2):
     assert 'pi' not in value
     assert pis == value['pis']
     assert value['schema_version'] == '3'
+
+
+def test_award_upgrade_2_3(upgrader, award_v3):
```
Comment (ian-whaling) [inline src/igvfd/tests/test_upgrade_award.py:18]: test_award_upgrade_2_3 should be test_award_upgrade_3_4

File: src/igvfd/tests/test_upgrade_biomarker.py [hunk 4]
```diff
@@ -5,3 +5,9 @@ def test_biomarker_upgrade_1_2(upgrader, biomarker_v1):
     value = upgrader.upgrade('biomarker', biomarker_v1, current_version='1', target_version='2')
     assert value['status'] == 'in progress'
     assert value['schema_version'] == '2'
+
+
+def test_biomarker_upgrade_1_2(upgrader, biomarker_v2):
```
Comment (ian-whaling) [inline src/igvfd/tests/test_upgrade_biomarker.py:10]: test_biomarker_upgrade_1_2 should be test_biomarker_upgrade_2_3

File: src/igvfd/tests/test_upgrade_document.py [hunk 5]
```diff
@@ -6,3 +6,9 @@ def test_document_upgrade_1_2(upgrader, document_v1):
     assert 'urls' not in value
     assert 'aliases' not in value
     assert value['schema_version'] == '2'
+
+
+def test_document_upgrade_1_2(upgrader, document_v2):
```
Comment (ian-whaling) [inline src/igvfd/tests/test_upgrade_document.py:11]: test_document_upgrade_1_2 should be test_document_upgrade_2_3

File: src/igvfd/tests/test_upgrade_phenotypic_feature.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import pytest
+
+
+def test_phenotypic_feature_upgrade_3_4(upgrader, phenotypic_feature_v1):
```
Comment (ian-whaling) [inline src/igvfd/tests/test_upgrade_phenotypic_feature.py:4]: test_phenotypic_feature_upgrade_3_4 should be test_phenotypic_feature_upgrade_1_2

File: src/igvfd/tests/test_upgrade_publication.py [hunk 7]
```diff
@@ -13,3 +13,9 @@ def test_publication_upgrade_1_2(upgrader, publication_v2):
     assert 'identifiers' not in value
     assert 'publication_identifiers' in value and value['publication_identifiers'] == ids
     assert value['schema_version'] == '3'
+
+
+def test_publication_upgrade_1_2(upgrader, publication_v3):
```
Comment (ian-whaling) [inline src/igvfd/tests/test_upgrade_publication.py:18]: test_publication_upgrade_1_2 should be test_publication_upgrade_3_4.
Unrelated but I think the above function should be renamed similarly as well.

File: src/igvfd/tests/test_upgrade_workflow.py [hunk 8]
```diff
@@ -9,3 +9,11 @@ def test_workflow_upgrade_1_2(upgrader, workflow_v1):
     assert value['schema_version'] == '2'
     assert 'publication_identifiers' in value and value['publication_identifiers'] == ids
     assert 'references' not in value
+
+
+def test_workflow_upgrade_1_2(upgrader, workflow_v2):
```
Comment (ian-whaling) [inline src/igvfd/tests/test_upgrade_workflow.py:14]: test_workflow_upgrade_1_2 should be test_workflow_upgrade_2_3

**Review by ian-whaling — CHANGES_REQUESTED** [review]

**Review by ian-whaling — APPROVED** [review]

**Pull request merged**